AwarenessApisSampleKotlin/app/src/main/java/com/android/example/awarenessapissamplekotlin/MainActivity.kt

```diff
@@ -37,6 +55,83 @@ class MainActivity : AppCompatActivity() {
 
     fun onClickSnapshot(view: View) {
         Log.d(TAG, "onClickSnapshot()")
+
+        if (activityRecognitionPermissionApproved()) {
+            requestSnapshot()
+        } else {
+            requestPermissionLauncher.launch(Manifest.permission.ACTIVITY_RECOGNITION)
+        }
+    }
+
+    private fun requestSnapshot() {
+        Log.d(TAG, "requestSnapshot()")
+
+        val task: Task<DetectedActivityResponse> =
+            Awareness.getSnapshotClient(this).detectedActivity
+
+        task.addOnCompleteListener { taskResponse: Task<DetectedActivityResponse> ->
+
+            if (taskResponse.isSuccessful) {
+                val detectedActivityResponse: DetectedActivityResponse = taskResponse.result
+                val activityRecognitionResult: ActivityRecognitionResult =
+                    detectedActivityResponse.activityRecognitionResult
+                Log.d(TAG, "Snapshot successfully retrieved: $activityRecognitionResult")
+                printSnapshotResult(detectedActivityResponse.activityRecognitionResult)
+            } else {
+                Log.d(TAG, "Data was not able to be retrieved: ${taskResponse.exception}")
+            }
+        }
+    }
+
+    private fun activityRecognitionPermissionApproved(): Boolean {
+        // Permission check for 29+.
+        return if (PlatformVersion.isAtLeastQ()) {
+            PackageManager.PERMISSION_GRANTED == ContextCompat.checkSelfPermission(
+                this,
+                Manifest.permission.ACTIVITY_RECOGNITION
+            )
+        } else {
+            true
+        }
+    }
+
+    private val requestPermissionLauncher: ActivityResultLauncher<String> =
```

Comment:
If we can, we can simplify to `private val requestPermissionLauncher =`

Reply:
I'd like to keep this one, as the variable name doesn't match the exact class (but I want to keep the var name so devs know I'm requesting permission). 

This can help with devs online who want may want to easily know the class to look up kdocs without downloading the sample.

Reply:
Ah, that's good to know about this tip to help devs know which class to lookup without downloading the sample. Will keep in mind for my future commits too.